Creating and styling form

annasvrv opened the pull request.

annasvrv committed:
Added images to the project

Created sprite svg for icons. Restructured images folders, fixed img formats.
Need to clean up remote repo.

annasvrv committed:
Removed file DS_Store

annasvrv committed:
Added images to the homepage

annasvrv committed:
remove redundant files

annasvrv committed:
remove redundant files

annasvrv committed:
remove redundant files

annasvrv committed:
Added icons and images to the pages

annasvrv committed:
Added images and icons to the page

annasvrv committed:
Added images and icons to the page

annasvrv committed:
Added images and icons to the page

annasvrv committed:
Removed unnecessary comments

annasvrv committed:
Added svg icon into sprite file

annasvrv committed:
Styled blog page

Added styles, images and icons to the page

annasvrv committed:
Added images and icons to the page

annasvrv committed:
Added styles and images on the page

annasvrv committed:
Added safe fonts, fixed fonts on the page

annasvrv committed:
Created newsletter form for homepage

annasvrv committed:
Created newsletter form

annasvrv committed:
Created newsletter form

annasvrv committed:
Created newsletter form

annasvrv committed:
Created newsletter form for the page.

Fixed pagination, added hover effect to the pages.

annasvrv committed:
Created forms

Created search form, filters and newsletter form

annasvrv committed:
Added forms and styled them

Created dropdown menu on hover.
Created newsletter form. Cleaned up comments, refactored some classes.

annasvrv committed:
Created forms and styled them.

Created header dropdown menu.
Created search form, filters form with checklists, added current filters display bar.
Created newsletter form in the footer.
Cleaned up comments and refactored some classes.

annasvrv committed:
Created forms and styled them.

Created header dropdown menu on hover.
Created footer newsletter form. Cleaned up.

annasvrv committed:
Created forms and styled them.

Added header dropdown menu on hover.
Created get in touch form, added patterns and validation.
Added footer newsletter form, cleaned up.

annasvrv committed:
Created forms and styled them.

Added header dropdown menu on hover.
Added footer newsletter form, cleaned up.

annasvrv committed:
Created forms and styled them.

Added header dropdown menu on hover.
Created search form, filters checkbox list.
Created feedback form, added patterns and validation.
Added footer newsletter form. Cleaned up.

annasvrv committed:
Created forms and styled them.

Added header dropdown menu on hover.
Updated hover effect border shine on cards.
Added footer newsletter form. Cleaned up.

annasvrv committed:
Created forms and styled them.

Added header dropdown menu on hover.
Added footer newsletter form. Cleaned up.

annasvrv committed:
Added icons svg

annasvrv committed:
Added styles for forms. Cleaned up.

annasvrv committed:
Added icons svg

annasvrv committed:
Fixes from PR4 - classes and css

Changed line-height to number value.
Fixed rgba with 1 opacity - changed to rgb.
Fixed classes names.

annasvrv committed:
Refactoring tags and classes

annasvrv committed:
fixed alt description, added img sizes, fixed box-sizing

annasvrv committed:
Added configuration file for GitHub Actions

per-aspera-ad-astra commented on index.html:
```diff
@@ -44,91 +44,76 @@
-          <li class="nav-menu-item">
-            <a href="index.html">HOME</a>
+          <li>
+            <a href="index.html" class="nav-menu-item">HOME</a>
```
Better use styles to uppercase words

annasvrv replied:
#fixed

per-aspera-ad-astra commented on css/style.css:
```diff
@@ -86,92 +107,132 @@ button {
 header {
   margin: 30px 0;
+  display: flex;
+  flex-direction: row;
```
it's default value, you can remove it

annasvrv replied:
#fixed

per-aspera-ad-astra commented on css/style.css:
```diff
@@ -86,92 +107,132 @@ button {
 
 /********************* Header styles *****************************/
 
 header {
```
better use class instead of tag

annasvrv replied:
#fixed

per-aspera-ad-astra commented on css/style.css:
```diff
@@ -86,92 +107,132 @@ button {
+  vertical-align: middle;
+  text-align: center;
+  line-height: 60px;
+  content: url(../img/icons/switch.svg);
```
Use quotes for url():
`url("../img/icons/switch.svg")`

annasvrv replied:
#fixed

per-aspera-ad-astra commented on css/style.css:
```diff
@@ -690,141 +818,275 @@ footer .media-btn {
-  padding-top: 20px;
-  padding-bottom: 20px;
+.nft-filter[open] .nft-filter-name::after {
+  content: "";
```
no need `content` here, you added this property above

annasvrv replied:
#fixed

per-aspera-ad-astra commented:
Some checkboxes not working: 
<img width="246" alt="Screenshot 2024-08-13 at 23 55 14" src="https://github.com/user-attachments/assets/37b0485f-962f-4dbb-87c8-0abc0e35123a">

per-aspera-ad-astra commented:
To much space between` textarea` and `button`:
<img width="455" alt="Screenshot 2024-08-14 at 00 02 10" src="https://github.com/user-attachments/assets/f96a9937-8fd3-4f59-8845-2f07030910ae">

per-aspera-ad-astra commented on index.html:
```diff
@@ -859,79 +911,88 @@ <h2 class="secondary-heading">FAQS</h2>
+            required
+            pattern="[a-z0-9._%+\-]+@[a-z0-9.\-]+\.[a-z]{2,}$"
+          />
+          <input class="btn-light btn-basic" type="submit" value="SUBSCRIBE" />
```
Why didn't you use tag `button` here? Your classes say it should be a button )

annasvrv replied:
What is the difference and what is better to use in practice? 

I thought `<input type="submit">` and `<button type="submit">` are the same or interchangeable. 

thanks in advance.

per-aspera-ad-astra replied:
`input type="submit"` and `button type="submit"` do the same things. But using button is better solution because it more flexible for styling - you can add images or icons inside, use pseudo elements.

annasvrv replied:
Thank you for explanation. 

#fixed

per-aspera-ad-astra commented:
Inputs change height if I start typing inside them
<img width="310" alt="Screenshot 2024-08-14 at 00 12 51" src="https://github.com/user-attachments/assets/86f64b9f-e435-424d-bb7b-cc0c05aa1b89">

per-aspera-ad-astra reviewed (changes requested):
Good job!

What I liked:

- Using github pages - it's convenient to review
- Task meet the requirement - forms implemented, all forms have validation, button have type attributes

Need to improve:

- Please, merge this pull pull request into your main branch before you start next task. It's a little more difficult to review the code if it contains changes from a previous task
- Fix provided comments, after that I can approve this PR

per-aspera-ad-astra commented on contact-us.html:
```diff
@@ -147,35 +138,90 @@ <h1 class="secondary-heading">CONTACT US</h1>
+        </p>
+        <form action="" method="post" class="get-intouch-form">
+          <fieldset class="form-block">
+            <label class="visually-hidden" for="name">Your full name:</label>
```
I'm not sure if this is needed here

annasvrv replied:
#fixed

per-aspera-ad-astra replied:
I told about hidden label :)

annasvrv replied:
I don't understand why no need label??? could you clarify please?

per-aspera-ad-astra replied:
It is invisible and does not affect the user’s interaction with the input, so I think it can be removed from the markup

annasvrv replied:
Hmm, we were taught that labels are required and should be visually hidden by design. 

https://css-tricks.com/html-inputs-and-labels-a-love-story/
 
So I think I won't change it till further clarification.

per-aspera-ad-astra replied:
Ok, let's leave it

annasvrv committed:
Fixed PR 5

Fixes contain text transform properties.
Url in quotes.
Check list added value, id, and for property.
Fixed margin, clean up.
Changed input to button.
Fixed border in inputs fields.
Added fix position to header. Now when scroll can see nav menu.

per-aspera-ad-astra commented on css/style.css:
```diff
@@ -109,16 +110,27 @@ button {
+  top: 0px;
+  left: 0;
+  width: 100%;
+  height: 108px;
```
You can move this value to variable and use it here and as `margin-top` on main block

annasvrv replied:
Do you mean 

`:root {
        --indent-var: 108px;
   }


.main {
  margin-top: var(--indent-var);
}`.      ?

per-aspera-ad-astra replied:
Yes, better choose name `--header-height` or something similar and use it for header too:
```
.header {
...
height: var(--header-height);
...
}
```

annasvrv replied:
#fixed

annasvrv committed:
added variable for header height.

per-aspera-ad-astra reviewed (approved):
Can be merged into main branch
